File: c/pomodoro.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#define true 1
#define false 0

void clear_terminal(void)
{
    system("clear");
}

void display_time(int x)
{
    int div = x / 60;
    int mod = x % 60;
    char minutes[3] = {(div / 10) + '0', (div % 10) + '0', '\0'};
    char seconds[3] = {(mod / 10) + '0', (mod % 10) + '0', '\0'};

    printf("%s:%s\n", minutes, seconds);
}

void run(void)
{
    int x = 0;

    while (x < (25 * 60)) {
        clear_terminal();
        printf("Pomodoro (25 minutes):\n");
        display_time(x);
        sleep(1);
        x += 1;
    }
}

void _pause(int duration)
{
    int x = 0;

    while (x < (duration * 60)) {
        clear_terminal();
        printf("Pause (%d minutes):\n", duration);
        display_time(x);
        sleep(1);
        x += 1;
    }
}

int main(void)
{
    int count_run = 0;

    while (true) {
        run();
        count_run += 1;
        if ((count_run > 0) && (count_run % 4 == 0)) {
            _pause(15);
        } else {
            _pause(5);
        }
    }
    return (0);
}

File: go/pomodoro.go
package main

import (
	"fmt"
	"time"
	"strconv"
	"os/exec"
	"os"
)

func clear_terminal() {
	cmd := exec.Command("clear")
	cmd.Stdout = os.Stdout
	cmd.Run()
}

func display_time(x int) {
	div := x / 60
	mod := x % 60
	var minutes, seconds string

	if div < 10 {
		minutes = "0" + strconv.Itoa(div)
	} else {
		minutes = strconv.Itoa(div)
	}

	if mod < 10 {
		seconds = "0" + strconv.Itoa(mod)
	} else {
		seconds = strconv.Itoa(mod)
	}

	fmt.Println(minutes + ":" + seconds)
}

func run() {
	x := 0

	for x < (25 * 60) {
		clear_terminal()
		fmt.Println("Pomodoro (25 minutes):")
		display_time(x)
		time.Sleep(1 * time.Second)
		x += 1
	}
}

func pause(duration int) {
	x := 0

	for x < (duration * 60) {
		clear_terminal()
		fmt.Println("Pause (" + strconv.Itoa(duration) + " minutes):")
		display_time(x)
		time.Sleep(1 * time.Second)
		x += 1
	}
}

func main() {
	count_run := 0

	for {
		run()
		count_run += 1
		if (count_run > 0) && (count_run % 4 == 0) {
			pause(15)
		} else {
			pause(5)
		}
	}
}
